content/algorithms/assortativity/correlation.md

```diff
@@ -0,0 +1,228 @@
+
+We may say two nodes are similar if they have the same/similar property i.e. a
+node $u$ is similar to $v$ if both of them have the same degree/nominal/numerical
+value of a property.
+Based on these properties we can have a different measure of assortativity for the network.
+
+### Assortativity coefficients
+
+Lets say we have a network $N$, $N = (V,E)$ and a poperty $P(v)$ for each node $v$.
+
+#### Mixing matrix
+
+Lets the property $P(v)$ take $P_0,P_1,...P_{k-1}$ distinct values on the network, 
+then Mixing matrix is matrix $M$ such that $M[i][j]=$ number of edges from
+nodes with property $P_i$ to $P_j$.
+We can normalize mixing matrix by diving by total number of ordered edges i.e.
+$ e = \frac{M}{|E|}$.
+
+Now define,
+
+$a[i]=$ proportion of edges $(u,v)$ such that $P(u)=P_i$
+
+$$ a[i] = \sum\limits\_{j}e[i][j] $$
+
+$b[i]=$ proportion of edges $(u,v)$ such that $P(v)=P_i$
+
+$$ b[i] = \sum\limits\_{j}e[j][i]$$
+
+$\sigma_a$ and $\sigma_b$ as standard deviation of
+$\{\ i\cdot a[i]\ |\ i \in 0...k-1\}$ and $\{ i\cdot b[i]\ |\ i \in 0...k-1\}$
+respectively.
+Note, after this will use subscript notation to denote indexing i.e. $P_i=P[i]$
+
+Then we can define the assortativity coefficient for this property based on the
+Pearson correlation coefficient.
+
+#### Attribute Assortativity Coefficient
+
+Here the property $P(v)$ is a nominal property assigned to each node.
+As defined above we calculate the normalized mixing matrix $e$ and from that we
+define attribute assortativity coefficient as below.
+
+$$ r = \frac{\sum\limits*{i}e*{ii} - \sum\limits*{i}a*{i}b*{i}}{1-\sum\limits*{i}a*{i}b*{i}} = \frac{Trace(e) - ||e^2||}{1-||e^2||}$$
+
+It is implimented as `attribute_assortativity_coefficient`.
+
+#### Numeric Assortativity Coefficient
+
+Here the property $P(v)$ is a _non-negative integer_ property assigned to each
+node and we assume that $P[i] = i$, and the defination of the normalized mixing
+matrix $e$, $\sigma_a$, and $\sigma_b$ are same as above.
+From these we define numeric assortativity coefficient as below.
+
+$$ r = \frac{\sum\limits*{i,j}i j(e*{ij} -a_i b_j)}{\sigma_a\sigma_b} $$
+
+It is implimented as `numeric_assortativity_coefficient`.
+
+#### Degree Assortativity Coefficient
+
+When it comes to measuring degree assortativity for directed networks we have
+more options compared to assortativity w.r.t a property because we have 2 types
+of degrees, namely in-degree & out-degree.
+Based on the 2 types of degrees we can measure $2 \times 2 =4$ different types
+of assortativity.
+
+    1. r(in,in) : Measures tendency of having a directed edge (u,v) such that, in-degree(u) = in-degree(v).
+    2. r(in,out) : Measures tendency of having a directed edge (u,v) such that, in-degree(u) = out-degree(v).
+    3. r(out,in) : Measures tendency of having a directed edge (u,v) such that, out-degree(u) = in-degree(v).
+    4. r(out,out) : Measures tendency of having a directed edge (u,v) such that, out-degree(u) = out-degree(v).
```

Comment:
Could you check the output at https://53-40210422-gh.circle-artifacts.com/0/site/_build/html/content/algorithms/assortativity/correlation.html#degree-assortativity-coefficient ?

Maybe we can use markdown specific list, the tabbed ordered list seems to be a creating a code box.